connector/src/main/java/org/geysermc/connector/utils/ChunkUtils.java

```diff
@@ -135,10 +135,12 @@ public static ChunkData translateToBedrock(GeyserSession session, Column column,
         BitSet pistonOrFlowerPaletteIds = new BitSet();
 
         boolean overworld = session.getChunkCache().isExtendedHeight();
+        int maxBedrockSectionY = (overworld ? MAXIMUM_ACCEPTED_HEIGHT_OVERWORLD : MAXIMUM_ACCEPTED_HEIGHT) >> 4;
 
         for (int sectionY = 0; sectionY < javaSections.length; sectionY++) {
-            if (yOffset < ((overworld ? MINIMUM_ACCEPTED_HEIGHT_OVERWORLD : MINIMUM_ACCEPTED_HEIGHT) >> 4) && sectionY < -yOffset) {
-                // Ignore this chunk since it goes below the accepted height limit
+            int bedrockSectionY = sectionY + (yOffset - ((overworld ? MINIMUM_ACCEPTED_HEIGHT_OVERWORLD : MINIMUM_ACCEPTED_HEIGHT) >> 4));
+            if (!(0 <= bedrockSectionY && bedrockSectionY < maxBedrockSectionY)) {
```

Comment:
Nitpickiest of nitpicks: can you invert the conditions here so we don't have this awkward `(!(` going on? Should be:
`0 > bedrockSectionY || bedrockSectionY >= maxBedrockSectionY`.

Reply:
(Also, is that last check correct? Seems like it'll cut off the final, highest chunk)

Reply:
The condition should look correct now with the highest chunk still intact.